### ISSUE TITLE
Minimize of a verity hash partition should work regardless of data partition options

### ISSUE THREAD

**Issue by mcassaniti** [systemd/systemd#27441]
### systemd version the issue has been seen with

commit b640e274a7c363a2b6394c9dce5671d9404d2e2a (v253+)

### Used distribution

Ubuntu 22.10

### Linux kernel version used

5.19.0-23-generic

### CPU architectures issue was seen on

x86_64

### Component

systemd-repart

### Expected behaviour you didn't see

If `systemd-repart` is used to populate a disk and also calculate the verity partition contents, it should also be able to minimize the verity partition regardless of the size options for the data partition. Unless I am incorrect, the data partition size and contents should be known by the time the verity partition is populated.

See below for my configuration. The disk image is build using `mkosi`.

### Unexpected behaviour you saw

`Minimize= set for verity hash partition but data partition does not set CopyBlocks= or Minimize=`

### Steps to reproduce the problem

Add the contents below to a repart directory.

```ini
# OS partition
[Partition]
Type           = root
Format         = ext4

SizeMinBytes   = 15G
SizeMaxBytes   = 15G

CopyFiles      = /

VerityMatchKey = os
Verity         = data
```

```ini
# Verify partition
[Partition]
Type           = root-verity

Minimize       = best

# Hash the data partition and populate with the hash contents
VerityMatchKey = os
Verity         = hash
```

Run the command below.
```shell
systemd-repart --empty=create --size=auto --dry-run=no --definitions <repart-directory> <image-output-file.img>
```

**Comment by mcassaniti** [comment]
The best option I can see is to run through `context_minimize()` for the data partition so that the hash and signature partitions get created using a stand-alone data partition. I'll see if I can get a rough patch together.

**Comment by DaanDeMeyer** [comment]
> If systemd-repart is used to populate a disk and also calculate the verity partition contents, it should also be able to minimize the verity partition regardless of the size options for the data partition. Unless I am incorrect, the data partition size and contents should be known by the time the verity partition is populated.

This is correct, but repart calculates all partition sizes up front, and that happens before the data partition is populated (unless it's minimized).

**Comment by DaanDeMeyer** [comment]
What you could do is run repart twice, first with `--exclude-partitions=verity,verity-sig --size=auto` to populate the data partition and then with just `--size=auto`. The second time the size and contents of the data partition will be known already so Minimize= will work for the verity hash partition.

**Comment by mcassaniti** [comment]
@DaanDeMeyer Take a look at #27443 which just might work.
